### PR TITLE
Fixed failing links after removing deprecated pages

### LINKED ISSUE #3443: Failing link check (Monday, September 16th)
The broken link check failed. Check [the workflow logs](https://github.com/timescale/docs/actions/workflows/daily-link-checker.yml) to identify the failing links.

### PR DESCRIPTION
**Pull request by atovpeko**
# Description

[Short summary of why you created this PR]

# Links

Fixes #[insert issue link, if any]

# Writing help

For information about style and word usage, see the [style guide](https://docs.timescale.com/about/latest/contribute-to-docs/)

# Review checklists

Reviewers: use this section to ensure you have checked everything before approving this PR:

## Subject matter expert (SME) review checklist

*   [ ] Is the content technically accurate?
*   [ ] Is the content complete?
*   [ ] Is the content presented in a logical order?
*   [ ] Does the content use appropriate names for features and products?
*   [ ] Does the content provide relevant links to further information?

## Documentation team review checklist

*   [ ] Is the content free from typos?
*   [ ] Does the content use plain English?
*   [ ] Does the content contain clear sections for concepts, tasks, and references?
*   [ ] Have any images been uploaded to the correct location, and are resolvable?
*   [ ] If the page index was updated, are redirects required
      and have they been implemented?
*   [ ] Have you checked the built version of this content?

### CODE REVIEW

File: about/release-notes/index.md [hunk 1]
```diff
@@ -23,10 +23,7 @@ click `Watch`, select `Custom` and then check `Releases`.
 
 </Highlight>
 
-This page contains the release notes from TimescaleDB [v2.10.0](#timescaledb2100-on-2023-02-21) 
-
-to [v2.15.1](#timescaledb2151-on-2024-05-28). For release notes for older versions, see the
-[past releases section][past-relnotes].
+This page contains the release notes from TimescaleDB [v2.10.0](#timescaledb2100-on-2023-02-21) to [v2.15.1](#timescaledb2151-on-2024-05-28).
```
Comment (billy-the-fish) [inline about/release-notes/index.md:26]: Local links don't work when the docs are published. It has to be a full link like [about-hypertables]: /use-timescale/:currentVersion:/hypertables/.

Comment (billy-the-fish) [inline about/release-notes/index.md]: Can you delete this page as well please, it is no longer used.

Comment (atovpeko) [reply about/release-notes/index.md]: Done

**Review by billy-the-fish — APPROVED** [review]
LGTM

**Pull request merged**